### DISCUSSION TITLE
Decision curve for decision tree model

### DISCUSSION THREAD

**Discussion by Meech81** [MSKCC-Epi-Bio/decisioncurveanalysis#4] [Q&A]
Hello,

I'm hoping for some advice please @ddsjoberg @VickersA 😁

Does it make sense to carry out a decision curve analysis for a decision tree model?

Below is an example I have taken from a worked example using the 'ctree' function in R (https://www.tutorialspoint.com/r/r_decision_tree.htm) to determine from a set of reading skills whether a person is a native speaker or not. The response variable is binary 'native speaker' yes or no and the model outcome is in the same format (i.e. binary yes / no). It is possible from these two variables to calculate the true positive and false positive achieved when using the decision tree. 

I have added on some code to create a decision curve. Since there is only one true positive and one false positive (as oppose to with a continuous model outcome / predicted probability where these would vary depending on cut-points) the only thing that changes when calculating the net benefit across the thresholds of the decision curve, is the weighting applied to the false positives. I'm also unsure of how the model outcome would relate to the decision curve thresholds.

Any advice would be appreciated.

Many thanks
Michelle 

```
library(tidyverse)
library(party)
library(caret)
library(dcurves)

# Print some records from data set readingSkills.
print(head(readingSkills))

# Create the input data frame.
input.dat <- readingSkills[c(1:105),]

# Give the chart file a name.
png(file = "decision_tree.png")

# Create the tree.
output.tree <- ctree(nativeSpeaker ~ age + shoeSize + score, data = input.dat)

# Plot the tree.
plot(output.tree)

# Predictions from tree
input.dat$pred <- predict(output.tree)

# Confusion matrix
cm <- confusionMatrix(input.dat$pred, input.dat$nativeSpeaker, positive = "yes")
cm$table

# creat dca object
input.dat$pred <- recode(input.dat$pred, "no" = 0, "yes" = 1)
dca_object <- dca(nativeSpeaker ~ pred, data = input.dat)
plot(dca_object)

# Extract data from dca object
dca_object_df <- dca_object$dca
summary(dca_object_df[dca_object_df $label=="pred",])
```

**Comment by VickersA** [comment]
Is this a question of how to run the analysis or how to interpret it?
